src/modules/Elsa.Workflows.Runtime/HostedServices/PopulateRegistriesHostedService.cs

```diff
@@ -24,8 +27,12 @@ public PopulateRegistriesHostedService(IServiceScopeFactory scopeFactory)
     public async Task StartAsync(CancellationToken cancellationToken)
     {
         using var scope = _scopeFactory.CreateScope();
-        var registriesPopulator = scope.ServiceProvider.GetRequiredService<IRegistriesPopulator>();
-        await registriesPopulator.PopulateAsync(cancellationToken);
+        var distributedLockProvider = scope.ServiceProvider.GetRequiredService<IDistributedLockProvider>();
+        await using (await distributedLockProvider.AcquireLockAsync("PopulateRegistries", TimeSpan.FromMinutes(10), cancellationToken))
```

Comment:
Do we want instances to possibly wait 10min before timing out?

Reply:
I'm thinking of it like this: each pod **must** go through this step, but they can only go through it one at a time. If it so happens to be that a pod B has to wait for pod A, there's no telling how long the process might take. Ideally, just a few milliseconds to a few seconds - perhaps one or two minutes if someone went crazy with the number of workflow definitions as activities. Regardless how long, pod B must wait. Obviously, 10 minutes is a very long time, which also means it should be relatively safe to set it to 10 minutes - if pod A takes longer than that, probably something isn't going very well over there. But instead of waiting for pod A indefinitely, the 10 minutes is an ultimate timeout to ensure pod B crashes with a timeout (which, by the way, could also happen if the shared resource to acquire a lock is unreachable).

That being said, it would be better for the user to configure this timeout, with a sensible default - which I think should be around 10 minutes, given that it is **very** unlikely that this step takes more than a couple of seconds, let alone 10 minutes, even with a large number of workflows in the system.

Reply:
I updated this section to leverage the Distributed Locking Timeout option which can be configured from the application itself.